radial_build.py

```diff
@@ -0,0 +1,337 @@
+
+                min_thickness = (min_lines + newlines) * min_line_height
+
+                thickness = min(max(layer['thickness'], min_thickness),
+                                self.max_thickness)
+
+                ax.add_patch(Rectangle(ll, thickness, height, facecolor=color,
+                                       edgecolor="black"))
+
+                # put the text in
+                centerx = ll[0] + thickness/2 + 1
+                centery = height/2
+
+                plt.text(centerx, centery, text, rotation="vertical",
+                         ha="center", va="center")
+
+                ll[0] += float(thickness)
+
+                total_thickness += thickness
+
+        ax.set_xlim(-1, total_thickness+1)
+        ax.set_axis_off()
+        plt.title(title)
+        plt.savefig(title.replace(' ', "") + '.png', dpi=200)
+        plt.close()
+
+    def get_toroidal_model(self, a, b, c):
+        """
+        Return toroidal model built using the build definition, contains
+        geometry and material information
+
+        Arguments:
+            a (float): major radius of torus in cm (around z axis)
+            b (float): minor radius (perpendicular to z) radial build offsets
+                from here
+            c (float): minor radius (parallel to z) radial build offsets from
+                here
+
+        Returns:
+            model (openmc model): Model containing materials and geometry
+                from the build dict.
+            cells (dict): dict mapping layer names to openmc cell instances in
+                the model object returned by this function.
+        """
+        # build surfaces
+        surfaces = {}
+
+        surfaces['plasma_surface'] = openmc.ZTorus(a=a, b=b, c=c)
+
+        for surface, surface_dict in self.build.items():
+            if surface_dict['thickness'] != 0:
+                b += surface_dict['thickness']
+                c += surface_dict['thickness']
+                surfaces[surface] = openmc.ZTorus(a=a, b=b, c=c)
+
+        # build regions
+        regions = {}
+
+        surf_list = list(surfaces.keys())
+
+        for i, surf in enumerate(surf_list):
+            if i == 0:
+                regions['plasma'] = -surfaces['plasma_surface']
+            else:
+                regions[surf] = -surfaces[surf] & +surfaces[surf_list[i-1]]
+
+        # build cells
+        cells = {}
+        materials = []
+
+        cells['plasma_cell'] = openmc.Cell(region=regions['plasma'],
+                                           name='plasma_cell')
+
+        for layer, layer_def in self.build.items():
+            if layer_def['thickness'] != 0:
+                try:
+                    cells[layer] = openmc.Cell(region=regions[layer],
+                                               name=layer,
+                                               fill=layer_def['material'])
+                    materials.append(layer_def['material'])
+                except KeyError as e:
```

Comment:
test input validity at beginning of function and then cleaner code with guaranteed valid input

Reply:
splitting these into separate classes makes me happy to just not check input for now, I will defer to your judgement however